### PR TITLE
Explicitly enable/disable datasharing on Throughtput test [11646]

### PR DESCRIPTION
**Pull request by richiware**

### CODE REVIEW

File: test/performance/latency/main_LatencyTest.cpp [hunk 1]
```diff
@@ -193,10 +117,12 @@ const option::Descriptor usage[] = {
...
     { FILE_R,        0, "f", "file",            Arg::Required,
       "  -f <arg>,  --file=<arg>             File to read the payload demands from." },
-    { DATA_SHARING,        0, "d", "data_sharing",            Arg::None,
-      "               --data_sharing        Enable data sharing feature." },
+    { DATA_SHARING,  0, "d", "data_sharing",    Arg::Enabler,
+      "               --data_sharing=[on|off]             Explicitly enable/disable data sharing feature." },
     { DATA_LOAN,        0, "l", "data_loans",            Arg::None,
       "               --data_loans          Use loan sample API." },
+    { SHARED_MEMORY,    0, "", "sharedmemory", Arg::Enabler,
+      "               --sharedmemory=[on|off]             Explicitly enable/disable shared memory transport." },
```
Comment (MiguelCompany) [inline test/performance/latency/main_LatencyTest.cpp:125]:
```suggestion
    { SHARED_MEMORY,    0, "", "shared_memory", Arg::Enabler,
      "               --shared_memory=[on|off]             Explicitly enable/disable shared memory transport." },
```

File: test/performance/throughput/main_ThroughputTest.cpp [hunk 2]
```diff
@@ -198,10 +120,12 @@ const option::Descriptor usage[] = {
...
     { UNKNOWN_OPT,   0, "",   "",               Arg::None,
       "\nNote:\nIf no demand or msg_size is provided the .csv file is used.\n"},
-    { DATA_SHARING,        0, "d", "data_sharing",            Arg::None,
-      "               --data_sharing        Enable data sharing feature." },
+    { DATA_SHARING,  0, "d", "data_sharing", Arg::Enabler,
+      "               --data_sharing=[on|off]             Explicitly enable/disable data sharing feature." },
     { DATA_LOAN,        0, "l", "data_loans",            Arg::None,
       "               --data_loans          Use loan sample API." },
+    { SHARED_MEMORY,    0, "", "sharedmemory", Arg::Enabler,
+      "               --sharedmemory=[on|off]             Explicitly enable/disable shared memory transport." },
```
Comment (MiguelCompany) [inline test/performance/throughput/main_ThroughputTest.cpp:128]:
```suggestion
    { SHARED_MEMORY,    0, "", "shared_memory", Arg::Enabler,
      "               --shared_memory=[on|off]             Explicitly enable/disable shared memory transport." },
```

**Review by MiguelCompany — APPROVED** [review]
LGTM

**Review by MiguelCompany — APPROVED** [review]

**Comment by richiware** [comment]
Failed tests not related. Merging..

**Pull request merged**